tests/framework/results/test_context.py

```diff
@@ -70,3 +67,116 @@ def test_add_stratification_raises(
     ctx = ResultsContext()
     with pytest.raises(expected_exception):
         raise ctx.add_stratification(name, sources, categories, mapper, is_vectorized)
+
+
+def _aggregate_state_person_time(self, x: pd.DataFrame) -> float:
+    """Helper aggregator function for observation testing"""
+    return len(x) * (28 / 365.35)
+
+
+@pytest.mark.parametrize(
+    "name, pop_filter, aggregator, additional_stratifications, excluded_stratifications, when",
+    [
+        (
+            "living_person_time",
+            'alive == "alive" and undead == False',
+            _aggregate_state_person_time,
+            [],
+            [],
+            "collect_metrics",
+        ),
+        (
+            "undead_person_time",
+            "undead == True",
+            _aggregate_state_person_time,
+            [],
+            [],
+            "time_step__prepare",
+        ),
+    ],
+    ids=["valid_on_collect_metrics", "valid_on_time_step__prepare"],
+)
+def test_add_observation(
+    name, pop_filter, aggregator, additional_stratifications, excluded_stratifications, when
+):
+    ctx = ResultsContext()
+    ctx._default_stratifications = ["age", "sex"]
+    assert len(ctx._observations) == 0
+    ctx.add_observation(
+        name,
+        pop_filter,
+        aggregator,
+        additional_stratifications,
+        excluded_stratifications,
+        when,
+    )
+    assert len(ctx._observations) == 1
+
+
+@pytest.mark.parametrize(
+    "name, pop_filter, aggregator, additional_stratifications, excluded_stratifications, when",
+    [
+        (
+            "living_person_time",
+            'alive == "alive" and undead == False',
+            _aggregate_state_person_time,
+            [],
+            [],
+            "collect_metrics",
+        ),
+    ],
+    ids=["valid_on_collect_metrics"],
+)
+def test_double_add_observation(
+    name, pop_filter, aggregator, additional_stratifications, excluded_stratifications, when
+):
+    """Tests a double add of the same stratification, this should result in one additional observation being added to
```

Comment:
Nit: This confused me at first b/c I thought you were saying that the second add should result in one additional (ie two total). Maybe something like this would be cleaner? "Tests a double add of the same stratification. Duplicate stratifications should be dropped and only one observation kept"

Reply:
Or "result in only one observation being added"

Reply:
Also, isn't this a duplicate add of the same *observation* not stratification?